CEDS 1187

ed0906 opened the pull request.

PatrykRudnicki commented on app/services/model/AutoCompleteItem.scala:
```diff
@@ -29,4 +29,8 @@ object AutoCompleteItem {
+
+  def fromNationalAdditionalCode(codes: List[NationalAdditionalCode]): List[AutoCompleteItem] = {
+    codes map (c => AutoCompleteItem(c.value, c.value))
+  }
```
Only style comment, but unnecessary parenthesis. It'll be good to keep convention like above for similar methods.

ed0906 replied:
Fixed

PatrykRudnicki commented on app/views/components/fields/field_multiple_values.scala.html:
```diff
@@ -0,0 +1,65 @@
+                    <th>
+                        <button
+                            class="button--secondary"
+                            name="@Remove.toString"
```
I'm not sure about it but it's better to ask than skip. I think we should wrap `Remove.toString` in parenthesis. Like "@{Remove.toString}"

ed0906 replied:
Fixed

ed0906 commented:
For anyone doing autocomplete stories, I've changed the name of the component to `field_autocomplete` and moved it into the `components.fields` package

ed0906 commented on app/utils/validators/forms/supplementary/ItemTypeValidator.scala:
```diff
@@ -67,14 +68,13 @@ object ItemTypeValidator extends Validator[ItemType] {
-      codes => codes.size <= nationalAdditionalCodesMaxAmount
-    )
+    "declaration.itemType.nationalAdditionalCode.error.maxAmount",
+    codes => codes.size <= nationalAdditionalCodesMaxAmount
```
Not sure this is technically needed as there isnt even 99 codes to use but I've left it in incase the csv grows in the future
